superset/db_engine_specs/base.py

```diff
@@ -737,7 +741,27 @@ def fetch_data(
         try:
             if cls.limit_method == LimitMethod.FETCH_MANY and limit:
                 return cursor.fetchmany(limit)
-            return cursor.fetchall()
+            data = cursor.fetchall()
+            description = cursor.description or []
+            column_type_mutators = {
+                row[0]: func
+                for row in description
+                if (
+                    func := cls.column_type_mutators.get(
+                        type(cls.get_sqla_column_type(cls.get_datatype(row[1])))
+                    )
+                )
+            }
+            if column_type_mutators:
```

Comment:
Typically decimal is a fixed point data structure but native `float` in Python is not fixed point data, so some drivers return a string to represent fixed point. `column_type_mutators` adds some extra transformation on the decimal column so the query performance might become lower than before.

From the screenshot before and after, I saw the `coltypes` field changed from `1` -> `0`, do we have some other approach just change the `coltypes` instead to mutate entire values of column?

Reply:
Good catch about the `coltypes`, let me investigate why that's gone missing.

Wrt to the query performance hit, another option is to make the frontend more resilient to receiving various types. In this case we could cast the value to `number` in the frontend if the coltype is `GenericDataType.NUMERIC`. However, I'd prefer to do this type of transformation in the backend to ensure that the plugins don't have to deal with all the nuances that SQLAlchemy throws our way.

I'd be curious to hear other peoples thoughts, @michael-s-molina @john-bodley any thoughts here?

Reply:
It hit me, that without this change, the linked issue won't be fixed, as CSV export happens purely in the backend. WIth this new pattern we could even normalize text based timestamps in the backend, eliminating the need for doing it in the frontend, further decoupling the quirks of the analytical databases from the downstream logic.

Reply:
Btw, I just noticed we already have logic like this in the codebase: https://github.com/apache/superset/blob/50535e427e51743e298d46009d51b08ef2380012/superset/db_engine_specs/ocient.py#L330-L334